ansible/roles/flannel/tasks/client.yml

```diff
@@ -4,15 +4,33 @@
   args:
         name: flannel
         state: latest
-  when: not is_atomic and not is_coreos
+  when: not is_atomic and not is_coreos and ansible_distribution != "Ubuntu"
+
+- name: Force flannel_source_type to github when packages are not available
+  set_fact:
+    flannel_source_type: "github-release"
```

Comment:
This is an unacceptable ansible pattern. flannel_source_type will not be fined on anything other than coreos/ubuntu and thus when you try to use it below this will cause the playbook to fail.

Instead, the very first thing you should do in this playbook is unconditionally set flannel_source_type to "package_manager" and then second overwirte it to "github-release" on the correct 2 systems.  Only run the {{ ansible_pkg_mgr }} if it is "package_manager" and run the github-release plays only if it is "github-release"